rtl/eclrtl/eclrtl.cpp

```diff
@@ -2350,27 +2451,85 @@ void rtlTrimVAll(unsigned & tlen, char * & tgt, const char * src)
         }
         i++;
     }
-    buffer[tlen] = 0;
     tgt = buffer;
 }
 
+void rtlTrimVWS(unsigned & tlen, char * & tgt, const char * src, bool left, bool all, bool right)
+{
+    rtlTrimWS(tlen, tgt, strlen(src), src, left, all, right);
+}
+
 void rtlTrimVUnicodeAll(unsigned & tlen, UChar * & tgt, const UChar * src)
 {
     rtlTrimUnicodeAll(tlen, tgt, rtlUnicodeStrlen(src), src);
 }
 
+void rtlTrimVUnicodeWS(unsigned & tlen, UChar * & tgt, const UChar * src, bool left, bool all, bool right)
+{
+    rtlTrimUnicodeWS(tlen, tgt, rtlUnicodeStrlen(src), src, left, all, right);
+}
+
 ECLRTL_API void rtlTrimUtf8All(unsigned &tlen, char * &tgt, unsigned slen, const char * src)
 {
     //Go via unicode because of possibility of combining accents etc.
-    rtlDataAttr temp1(slen*sizeof(UChar));
-    rtlUtf8ToUnicode(slen, temp1.getustr(), slen, src);
+    rtlDataAttr temp1;
+    rtlUtf8ToUnicodeX(slen, temp1.refustr(), slen, src);
```

Comment:
minor: slightly questionable to use slen as the target for the length as well as the source - but the following code is correct (uses the updated value of slen which might possibly be larger than the original in the presence of surrogates)